src/commands/test/test_runner.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+from dataclasses import dataclass
+from logging import getLogger
+from pathlib import Path
+from typing import List, Optional
+
+from starkware.starknet.services.api.contract_definition import ContractDefinition
+from starkware.starkware_utils.error_handling import StarkException
+
+from src.commands.test.starkware_patch import CheatableSysCallHandler
+from src.commands.test.test_cases import BrokenTestFile, FailedTestCase, PassedTestCase
+from src.commands.test.test_environment_exceptions import ReportedException
+from src.commands.test.test_environment import TestEnvironment
+from src.commands.test.test_subject_queue import TestSubject, TestSubjectQueue
+from src.utils.modules import replace_class
+from src.utils.starknet_compilation import StarknetCompiler
+
+current_directory = Path(__file__).parent
+
+logger = getLogger()
+
+
+class TestRunner:
+    include_paths: Optional[List[str]] = None
+    _collected_count: Optional[int] = None
+
+    def __init__(
+        self,
+        queue: TestSubjectQueue,
+        include_paths: Optional[List[str]] = None,
+    ):
+        self.queue = queue
+        self.include_paths = []
+
+        if include_paths:
+            self.include_paths.extend(include_paths)
+
+    @dataclass
+    class WorkerArgs:
```

Comment:
I can bring it back, but I rather have type-safe and easy to maintain rather than pytonic code.

Reply:
For this one acctualy im okay with this class, because it's fir Scheduler and i didn't know at the time I wrote this comment. You can change `starmap` to `map` though.